test-integ/peering_commontopo/main_test.go

```diff
@@ -2,33 +2,25 @@ package peering
 
 import (
 	"flag"
+	"os"
 	"testing"
 )
 
-var FlagNoReuseCommonTopo *bool = flag.Bool("no-reuse-common-topo", false,
-	"run tests that can use the common topo in separate instances")
-
 type commonTopoSuite interface {
 	testName() string
 	setup(*testing.T, *commonTopo)
 	test(*testing.T, *commonTopo)
 }
 
-// Tests compatible with commonTopo should implement commonTopoSuite and be added
-// to the commonTopoSuites slice inside.
-//
-// func setup is executed in serial with others. They should ensure any resources
-// added to ct.Cfg et all do not collide with other resources (e.g. with a prefix)
-//
-// func test is executed in parallel with others in a subtest. (test() should not
-// call t.Parallel() itself.)
-func TestAllCommonTopoSuites(t *testing.T) {
-	if *FlagNoReuseCommonTopo {
-		t.Skip("NoReuseCommonTopo set")
-	}
-	if allowParallelCommonTopo {
-		t.Parallel()
+func CommonTopologyTest(t *testing.T) {
+	t.Helper()
+	if os.Getenv("COMMON_TOPOLOGY") != "true" {
```

Comment:
Huh. I thought you were going to try using build tags, not an environment variable?

Reply:
Personally I prefer a proper, named, documented boolean flag to an env var from a UX perspective

Reply:
Enh, maybe worth a discussion with the larger group, but with a brief scan, I think I disagree with most of that author's advice 😉

Reply:
With that being said, I do like breaking out the test setup like @nfi-hashicorp did in his other experimental branch. Basically, moving more of the logic of common topo setup into the helper. 

So overall I think I would prefer a combination.

* Use flags for detection of common topo usage
* Move all of the logic around building out the topology (and whether it should use the common infra) into a helper that can be called from all the tests which could share a topology.